client-core/src/service/wallet_service.rs

```diff
@@ -224,7 +217,7 @@ where
         staking_key: &PublicKey,
     ) -> Result<()> {
         self.modify_wallet(name, enckey, move |wallet| {
-            wallet.staking_keys.insert(staking_key.clone());
+            wallet.staking_keys.push(staking_key.clone());
```

Comment:
We now need to check here if given `staking_key` already exists and don't push if it does.